common/src/main/kotlin/org/neo4j/connectors/kafka/utils/ListUtils.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [https://neo4j.com]
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.connectors.kafka.utils
+
+object ListUtils {
+  infix fun <T> List<T>.equalsIgnoreOrder(other: List<T>): Boolean {
+    return this.size == other.size && this.toSet() == other.toSet()
```

Comment:
Probally neat pick: maybe `this.size == other.size && this.containsAll(other)` will be slightly better?

Reply:
wouldn't you run in edge cases like (in pseudo-code) `["a", "b"].equalsIgnoreOrder(["a", "a"])`?

Reply:
oh but I realize the current implementation is also sorta subject to fun edge cases like `["a", "a", "b"].equalsIgnoreOrder(["a", "b", "b"])`.
I guess the only reliable way is to enforce that `T` is `Comparable`, sort the two lists and compare them element-wise?

Reply:
```kotlin
    if (this.size != second.size) {
        return false
    }
    for (i in this.indices) {
        if (this[i] != second[i]) {
            return false
        }
    }
    return true
```

Reply:
What about this one?

```
return this.size == other.size &&
        this.groupingBy { it }.eachCount() == other.groupingBy { it }.eachCount()
```

Reply:
If we are writing a generic purpose method, I agree with @fbiville. We don't care about duplicates both `toSet()` and `containerAll` look tactically alright we rename the method. As far as I know, the same property cannot be a part of the key twice, so this edge case may not be realistic. We might need to add extra validation for duplicates.

Reply:
Maybe it's a valid case for a LinkedHashSet? 🤡

Reply:
That's the problem with "utils" functions. I'd expect them to be generic.
If they're not, I'd suggest co-locating them to the call site.

Reply:
yeah my opinion is to keep it as utils function so I'm gonna implement the solution which covers all possible cases.